tests/test_generator.py

```diff
@@ -0,0 +1,75 @@
+from deepinterpolation.generic import JsonSaver, ClassLoader
+import os
+import pathlib
+
+
+def test_generator_tif_creation(tmp_path):
+
+    generator_param = {}
+
+    # We are reusing the data generator for training here.
+    generator_param["type"] = "generator"
+    generator_param["name"] = "SingleTifGenerator"
+    generator_param["pre_post_frame"] = 30
+    generator_param["pre_post_omission"] = 0
+    generator_param[
+        "steps_per_epoch"
+    ] = -1
+
+    generator_param["train_path"] = os.path.join(
+        pathlib.Path(__file__).parent.absolute(),
+        "..",
+        "sample_data",
+        "ophys_tiny_761605196.tif",
+    )
+
+    generator_param["batch_size"] = 5
+    generator_param["start_frame"] = 0
+    generator_param["end_frame"] = 99
+    generator_param[
+        "randomize"
+    ] = 0
+
+    path_generator = os.path.join(tmp_path, "generator.json")
```

Comment:
similarly `tmp_path` is a `pathlib.Path` object, so this could just be:
```
path_generator = str(tmp_path / "gemerator.json")
```

Reply:
Will get familiar with PathLib a little more. Thanks for the tip. I just did a bit of reading around it